collect_app/src/main/java/org/odk/collect/android/tasks/SavePointTask.java

```diff
@@ -41,10 +43,12 @@
 
     private final SavePointListener listener;
     private int priority;
+    private FormIndex formIndex;
 
-    public SavePointTask(SavePointListener listener) {
+    public SavePointTask(SavePointListener listener, FormIndex formIndex) {
```

Comment:
It feels unfortunate to give this another responsibility that's not really related and to have to deal with null `formIndex`? Could it be its own `SaveFormIndexTask` that's only scheduled when needed?

Reply:
In my opinion, FormIndex is a part of savepoint so it should be saved in the same task. I decided to leave it and we can refactor the code in the future if you really want (an if you can explain why :smile: )

Reply:
It's a single responsibility thing. When I start seeing parameters added to constructors, `null`s being passed in and conditionals both in the class and in the clients it feels like we've gone from SavePointTask to SavePointAndIndexTask which feels like a red flag. Again, I understand all of this code is messy to start with so it's hard to know where to try to be principled. @jknightco would love your opinion on this.

Reply:
@jknightco what do you think of introducing a simple `SaveFormIndexTask` that just saves the index and is executed in the one necessary case? Is there another simple alternative you prefer?

Reply:
Sounds like a good call to me.